states/wyoming/parse_pdf.py

```diff
@@ -0,0 +1,191 @@
+
+    formatted_rows.append(first_county)
+    formatted_rows.append(second_county)
+
+  # Remove last nonexistent county.
+  formatted_rows = formatted_rows[:-1]
+
+  return formatted_rows
+
+
+def generate_county_dict_list(formatted_rows):
+  counties = []
+
+  for formatted_row in formatted_rows:
+    county = defaultdict(list)
+    county['county'] = formatted_row['title'].strip(' \n')[:-len(' Clerk')]
+    county['locale'] = county['county']
+    county['address'] = ' '.join(formatted_row['address'])
+    # Remove space between '-' in zipcode.
+    if '-' in county['address']:
+      address_string = county['address']
+      dash_index = address_string.index('-')
+      if address_string[dash_index - 1] == ' ':
+        address_string = address_string[:dash_index - 1] + address_string[dash_index:]
+      if address_string[dash_index + 1] == ' ':
+        address_string = address_string[:dash_index] + address_string[dash_index + 1:]
+      county['address'] = address_string
+
+    full_email = ''
+    for email_line in formatted_row['email']:
+      if 'Email' in email_line:
+        continue
+      full_email += email_line
+      if any(suffix in email_line for suffix in email_suffixes):
+        county['emails'].append(full_email)
+        full_email = ''
+
+    full_phone_number = ''
+    for phone_number_line in formatted_row['phone_number']:
+      phone_number_line = phone_number_line.strip('Ph. ')
+      # One phone number as 'or' in it...
+      if 'or' in phone_number_line:
+        phone_number_line = phone_number_line[:phone_number_line.index('or')]
+      # Some phone numbers span multiple lines.
+      full_phone_number += phone_number_line
+    # Manually add '-' since some phone number span multiple lines.
+    full_phone_number = full_phone_number.replace('.', '')
+    full_phone_number = (full_phone_number[:3]
+                         + "-"
+                         + full_phone_number[3:6]
+                         + "-"
+                         + full_phone_number[6:])
+    county['phones'].append(full_phone_number.replace('.', '-'))
+
+    for fax_line in formatted_row['fax']:
+      fax_line = fax_line.strip('Fax ')
+      county['faxes'].append(fax_line.replace('.', '-'))
+      
+    counties.append(county)
+
+  return counties
+
+
+def main():
+  with open(dir_path(__file__) + '/results/wyoming_contact.pdf', 'rb') as fh:
+    pdf_reader = PyPDF2.PdfFileReader(fh)
+    text = ''
+    for page_num in tqdm(range(pdf_reader.numPages)):
+      text += pdf_reader.getPage(page_num).extractText()
+
+  # Remove Page # of #
+  text = re.sub(r"\sPage\s\n\d\sof\s\n\d\s", "", text)
+
+  lines = make_lines(text)
+
+  # Remove first couples of lines which are PDF header.
+  lines = lines[9:]
+
+  rows = group_rows(lines)
+
+  grouped_rows = group_counties(rows)
+
+  counties = generate_county_dict_list(grouped_rows)
+
+  with open(dir_path(__file__) + '/results/records.noemail.json', 'w') as fh:
```

Comment:
Why do we need to save this twice?  Let's just save the official version in the public folder.